tests/unittest_brain.py

```diff
@@ -1580,10 +1683,6 @@ def __init__(self, value):
         """
         )
         inferred = next(node.infer())
-        assert len(inferred.slots()) == 0
```

Comment:
@Pierre-Sassoulas this test failed after I removed the additional check: `len(inferred.slots())` is 1 at this line. It seems the caching behaviour being tested no longer applies, since `mro` is called earlier, so the bases of the ClassDef are inferred earlier as well.

I'm not sure if this is good or bad, but it's definitely a change in behaviour. This functionality was introduced in #931, so you might want to check that PR and with the author?

Reply:
@cdce8p was the slots length supposed to be zero in the following code ? Do you remember what could be the problem if it's not ?
```python
from typing import Generic, TypeVar

T = TypeVar('T')


class A(Generic[T]):
    __slots__ = ['value']
    
     def __init__(self, value):
         self.value = value
```

Reply:
If I remember correctly, it's safe to remove that part of the check. As @david-yz-liu already mentioned, it was testing the, in that case invalid, caching of `slots`. Previously, `slots` would have been empty as the base wouldn't have been inferred as `Generic` just yet. What's important is that `slots` eventually contains the correct value, as tested a few lines below.

It seems as this behavior change is a side effect of this PR.